### ISSUE TITLE
False positive specific patterns in rule file_name

### ISSUE THREAD

**Issue by Dschee** [realm/SwiftLint#2417]
When I configure the rule `file_name` with a `suffix_pattern` which looks like this:

```
file_name:
  suffix_pattern: "Extensions?|\\+.*"
```

then my file `BoolExtension` is correctly handled. But when I add a test for that extension file called `BoolExtensionTests` then I get a violation warning there, although there shouldn't be one.
